calvinpete/Glokeeper-backend
# #83296631 Users should be able to create, fetch, update and delete a property

### What does this PR do?
This PR is to create the Property endpoint to enable users create, delete, read and update properties

### Description of Task to be completed?
Properties form the foundation of our application and introduce us to CRUD which is the building block of most web applications.

### How should this be manually tested?
- Fetch this branch in your terminal by running the command `git fetch origin ft-crud-property-183296631`
- Check into the branch by running `git checkout ft-crud-property-183296631`
- Activate your virtual environment by running `source venv/bin/activate`
- Install dependencies by running `pip install -r requirements.txt`
- Migrate by running `python manage.py migrate`
- Run `python manage.py runserver`
- Open postman to test the endpoints as shown in the screen shots below

## POST
<img width="1007" alt="post" src="https://user-images.githubusercontent.com/36506567/190887189-cf8e0332-2cc2-4a4d-a527-0951ca923973.png">


## PUT
<img width="1007" alt="put" src="https://user-images.githubusercontent.com/36506567/190887204-c53064f8-92bc-4254-8ece-90f541d35684.png">

## PATCH
<img width="1007" alt="patch" src="https://user-images.githubusercontent.com/36506567/190887211-5026645b-4183-47f4-88b9-d410e048f527.png">

## GET
<img width="1007" alt="fetch all" src="https://user-images.githubusercontent.com/36506567/190887201-6f5afc0f-875b-4dc3-8171-68f4212df819.png">

## GET
<img width="1007" alt="fetch 1" src="https://user-images.githubusercontent.com/36506567/190887208-c14985de-048b-4433-a0a2-c97d0359a015.png">

## DELETE
<img width="1007" alt="delete" src="https://user-images.githubusercontent.com/36506567/190887196-82aad69b-13f8-4ed1-b54f-5c9453c5ed31.png">

### What are the relevant pivotal tracker stories?
[#183296631](https://www.pivotaltracker.com/story/show/183296631)
